presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1149,6 +1149,26 @@ private void testInsertPartitionedTableExistingPartition(Session session, HiveSt
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testNullPartitionValues()
+    {
+        assertUpdate("" +
+                "CREATE TABLE test_null_partition (test VARCHAR, part VARCHAR)\n" +
+                "WITH (partitioned_by = ARRAY['part'])");
+
+        assertUpdate("INSERT INTO test_null_partition VALUES ('hello', 'test'), ('world', null)", 2);
+
+        assertQuery(
+                "SELECT * FROM test_null_partition",
```

Comment:
redundant?

Reply:
I thought it was useful to have a SQL test to make sure null partition values round trip (since Hive has the special `__HIVE_DEFAULT_PARTITION__` encoding for them).

Reply:
oh, ok, then this is not just a test for `$partitions`, but more general, for null in part key. Fine then.

Btw, how hive handle the literal value `'__HIVE_DEFAULT_PARTITION__'` in the part key?

Reply:
Correct, it's a general smoke test for null partitions.

Good question, I haven't tested it. My guess is it gets turned into a null (no escaping).